scripts/ci-helper.sh

```diff
@@ -163,48 +83,56 @@ function build() {
     ldflags="-s -w "
   fi
 
-  ldflags="${ldflags}-X github.com/hashicorp/vault/version.Version=$version -X github.com/hashicorp/vault/version.GitCommit=$revision -X github.com/hashicorp/vault/version.BuildDate=$build_date"
+  # if building locally with enos - don't need to set version/prerelease/metadata as the default from version_base.go will be used
+  ldflags="${ldflags} -X github.com/hashicorp/vault/version.GitCommit=$revision -X github.com/hashicorp/vault/version.BuildDate=$build_date"
+
+  if [[ ${BASE_VERSION+x} ]]; then
+    msg="${msg}, base version ${BASE_VERSION}"
+    ldflags="${ldflags} -X github.com/hashicorp/vault/version.Version=$BASE_VERSION"
+  fi
+
+  if [[ ${PRERELEASE_VERSION+x} ]]; then
+    msg="${msg}, prerelease ${PRERELEASE_VERSION}"
+    ldflags="${ldflags} -X github.com/hashicorp/vault/version.VersionPrerelease=$PRERELEASE_VERSION"
+  fi
 
-  if [ -n "$prerelease" ]; then
-    msg="${msg}, prerelease ${prerelease}"
-    ldflags="${ldflags} -X github.com/hashicorp/vault/version.VersionPrerelease=$prerelease"
+  if [[ ${VERSION_METADATA+x} ]]; then
+    msg="${msg}, metadata ${VERSION_METADATA}"
+    ldflags="${ldflags} -X github.com/hashicorp/vault/version.VersionMetadata=$VERSION_METADATA"
   fi
 
-  if [ -n "$metadata" ]; then
-    msg="${msg}, metadata ${metadata}"
-    ldflags="${ldflags} -X github.com/hashicorp/vault/version.VersionMetadata=$metadata"
+  if [ ${GO_TOOLCHAIN} == "fips" ]; then
+    mkdir .tools
+    cd .tools
+    curl --show-error --silent --fail --user "${ARTIFACTORY_USER}:${ARTIFACTORY_TOKEN}" --request GET \
```

Comment:
We should check to make sure this hasn't changed. I saw recent pull requests that were changing the fips toolchain setup. We'll want to make sure we don't accidentally undo.